concordium-std-derive/src/lib.rs

```diff
@@ -0,0 +1,188 @@
+extern crate proc_macro;
+
+use concordium_contracts_common::{
+    AccountAddress, ContractAddress, ModuleReference, PublicKeyEcdsaSecp256k1, PublicKeyEd25519,
+    SignatureEcdsaSecp256k1, SignatureEd25519,
+};
+use proc_macro::TokenStream;
+use std::str::FromStr;
+use syn::{parse_macro_input, LitStr};
+
+/// Procedural macro for instantiating account addresses.
+/// Input must be a valid base58-encoding.
+#[proc_macro]
+pub fn account_address(item: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(item as LitStr);
+    let str = input.value();
+
+    let addr_bytes = match AccountAddress::from_str(&str) {
+        Ok(addr) => addr.0,
+        Err(e) => {
+            let err = syn::Error::new(input.span(), format!("Invalid account address: {}", e));
+            return err.to_compile_error().into();
+        }
+    };
+
+    match format!("AccountAddress({:?})", addr_bytes).parse() {
+        Ok(o) => o,
+        Err(e) => {
+            let err = syn::Error::new(input.span(), format!("LexError: {}", e));
+            err.to_compile_error().into()
+        }
+    }
```

Comment:
So we can't use `quote!` to construct this? Something like
```
quote!(AccountAddress(#addr_bytes)
```

I don't really like parsing this thing again.